server/chroma/src/routes/v1/photo/create.rs

```diff
@@ -203,11 +211,29 @@ fn resize_and_save(
             }
         };
 
-        match photo.set_quality_created(quality, true).await {
-            Ok(_) => {}
+        // Verify, this also puts it in the cache, nice speedup for later
+        trace!("Checking if uploaded image actually works on AWS");
+
+        let url = match photo.photo_to_proto_url(&engine, &quality).await {
+            Ok(p) => match p.data.unwrap().response.unwrap() {
+                Response::Url(v) => v,
+                _ => panic!("Invalid response type for the 'URL' method"),
+            },
             Err(e) => {
-                warn!("Failed to set quality created flag for photo: {e}");
+                warn!("Photo {} with quality {} was not created successfully, or another error occurred: {e}", photo_id, quality);
+                return;
             }
+        };
+
+        // Fetch the photo
+        let ok = reqwest::Client::new().get(url).send().await.is_ok();
```

Comment:
This will always succeed if the endpoint is reachable as `is_ok()` only checks for the request successfully being executed, not whether it was a `200 OK`. For that, you can use `error_for_status()`: https://docs.rs/reqwest/latest/reqwest/struct.Response.html#method.error_for_status

Reply:
Oh damn how did I forget